file_converter/Program.cs

```diff
@@ -2,94 +2,73 @@
 using System.Linq;
 
 namespace file_converter
-{   
+{
     class Program
     {
+        private static string[] SUPPORTED_FORMATS = { "txt", "csv", "json", "xml" };
+        private static string GetFileExtension(string filePath)
+        {
+            string[] s = filePath.Split("."); 
+            return (s[s.Length-1]);
+        }
+        private static IFileWrapper BuildFileWrapper(string filePath)
+        {
+            string ext = GetFileExtension(filePath);
+            switch (ext)
+            {
+                case "txt":
+                    return new fileTxt(filePath);                   
+                case "csv":
+                    return new fileCsv(filePath);                   
+                case "json":
+                    return new fileJson(filePath);                   
+                case "xml":
+                    return new fileXml(filePath);
+            }
+            return new fileTxt(filePath);
+        }
         static void Main(string[] args)
         {
+            string firstArgExt = GetFileExtension(args[0]);
+            string secondArgExt = GetFileExtension(args[1]);
+            if (!SUPPORTED_FORMATS.Contains(firstArgExt) || !SUPPORTED_FORMATS.Contains(secondArgExt))
+            {
+                Console.WriteLine("Can`t convert such type of file :C");
+                return;
+            }
```

Comment:
At this step you don't know how many arguments are really present (there are no checks present before this fragment). You should move this fragment below the line 43.

Also, you don't have to check the extensions at this step. If any of the file paths have an invalid extension then `BuildFileWrapper` method will throw an exception.